src/code_manager/model/JsGenerator.js

```diff
@@ -55,25 +55,29 @@ export default Backbone.Model.extend({
       const mapType = this.mapJs[type];
 
       if (mapType.props) {
-        code += `
+        if (mapType.code && mapType.code.trim()) {
+          code += `
           var props = ${JSON.stringify(mapType.props)};
           var ids = Object.keys(props).map(function(id) { return '#'+id }).join(',');
           var els = document.querySelectorAll(ids);
           for (var i = 0, len = els.length; i < len; i++) {
             var el = els[i];
             (${mapType.code}.bind(el))(props[el.id]);
           }`;
+        }
       } else {
         // Deprecated
-        const ids = '#' + mapType.ids.join(', #');
-        code += `
+        if (mapType.code && mapType.code.trim()) {
```

Comment:
It's quite safe to assume that `mapType.code` will always be defined and trimmed (from `model.getScriptString`).
In this case, I also think we can avoid duplicating the same logic by simply checking on top
```js
...
if (!mapType.code) continue;

if (mapType.props) {
...
```

Reply:
Thanks for taking time to explain these concepts, much appreciated. I have a question regarding implementation of `isFunctionEmpty`. I've never needed this before and it seems to be a [complicated topic](https://stackoverflow.com/questions/7960335/javascript-is-given-function-empty).

After the first assert (`if (!mapType.code) continue;`) would asserting `isFunctionEmpty(mapType.code)` via this be safe?
```js
function isFunction(fn) {
  const content = fn.toString().match(/\{([\s\S]*)\}/m)[1]; // content between first and last { }
  return content.replace(/^\s*\/\/.*$/mg,'').length > 0; // remove comments
}
```
Can't we do something like minify the script with `uglify` and then check the length? That would also get us rid of adding newlines everytime in the old approach since https://github.com/artf/grapesjs/pull/3300 due to a rather edge-case scripts containing comments on the first line. Compressing emitted js would a great feature but it might be out of scope of this PR.

Reply:
The `isFunctionEmpty(mapType.code)` should be added inside `if (mapType.props) {` as the old approach tries already to clean the function signature.
In all honesty, I wouldn't spend too much time and effort on that check. Your `isFunctionEmpty` example is already cool and more than enough, I wouldn't even care about removing comments.
The minifying option is for sure a nice thing to have but I always thought about them as an extension/plugin for code generators but I still need to refactor and improve the API of this module to support such functionality.